android/jni/com/mapswithme/maps/MapManager.cpp

```diff
@@ -94,6 +95,13 @@ Java_com_mapswithme_maps_downloader_MapManager_nativeIsLegacyMode(JNIEnv * env,
   return !version::IsSingleMwm(GetStorage().GetCurrentDataVersion());
 }
 
+// static native boolean nativeNeedMigrate();
+JNIEXPORT jboolean JNICALL
+Java_com_mapswithme_maps_downloader_MapManager_nativeNeedMigrate(JNIEnv * env, jclass clazz)
+{
+  return platform::migrate::NeedMigrate();
```

Comment:
Я все равно не понимаю: эта проверка просто что-то проверяет в сеттингах. Когда и как это что-то выставляется? Почему настройка в сеттингах - это замена полноценной проверке? А если настройки забэкапились и восстановились? Или приложение переустановилось?
Единственный надежный способ проверки - это наличие или отсутствие новых mwm файлов, разве нет?

СС @bykoianko @ygorshenin 

> On Mar 25, 2016, at 17:32, Alexander Marchuk <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/MapManager.cpp <https://github.com/mapsme/omim/pull/2512#discussion_r57437723>:
> 
> > @@ -94,6 +95,13 @@ Java_com_mapswithme_maps_downloader_MapManager_nativeIsLegacyMode(JNIEnv * env,
> >    return !version::IsSingleMwm(GetStorage().GetCurrentDataVersion());
> >  }
> >  
> > +// static native boolean nativeNeedMigrate();
> > +JNIEXPORT jboolean JNICALL
> > +Java_com_mapswithme_maps_downloader_MapManager_nativeNeedMigrate(JNIEnv * env, jclass clazz)
> > +{
> > +  return platform::migrate::NeedMigrate();
> Это обсудили с Сергеем Е. В этом месте нужна быстрая проверка без поднятия фреймворка.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2512/files/2ed1aa65936a4f781a177777c26bb277038f0b0b#r57437723>